loftr_Code.py

```diff
@@ -0,0 +1,58 @@
+# Необходимые модули
+from pathlib import Path
+import os
+from hloc import extract_features, match_features, match_dense, reconstruction, visualization, pairs_from_retrieval
+from hloc.visualization import plot_images, read_image
+from hloc.utils import viz_3d
+
+# Снимки храма
+images = Path('datasets/South Building/')
+
+
+# Вывод
+outputs = Path('outputs/')
+outputs_loftr = outputs / 'LoFTR'
+outputs_netvlad = outputs / 'netVLAD'
+print(os.path.abspath(outputs))
+
+os.mkdir(outputs)
+os.mkdir(outputs_loftr)
+os.mkdir(outputs_netvlad)
+
+# Пары изображений (файл .txt)
+sfm_pairs = outputs_netvlad / 'pairs-sfm.txt'
+
+# Structure from Motion вывод
+sfm_dir_loftr = outputs_loftr / 'sfm'
+
+
+# Признаки
+features_loftr = outputs_loftr / 'features.h5'
+
+# Мэчи
+matches_loftr = outputs_loftr / 'matches.h5'
+
+# Retrieval
+retrieval_conf = extract_features.confs['netvlad']
+
+# Параметры моделей 
+# LoFTR:
+LoFTR_conf = match_dense.confs['loftr']
+
+
+
+# Изображения (10)
+references = [p.relative_to(images).as_posix() for p in (images).iterdir()]
+
+# Вывод
+print(len(references), "mapping images")
```

Comment:
Use python string formatting like this https://realpython.com/python-string-formatting/#3-string-interpolation-f-strings-python-36